src/panoptes/pocs/focuser/birger.py

```diff
@@ -249,36 +253,32 @@ def _send_command(self, command, response_length=None):
         # Clear the input buffer in case there's anything left over in there.
         self._serial_port.reset_input_buffer()
```

Comment:
This should be done before each time you send the command, i.e. you should move this line inside the loop just before the `self._serial_io.write(command + '\r')` line.

If you don't clear the buffer each time you might end up reading part of the response from a previous failed attempt to send the command, causing the response validation to fail unnecessarily.